lms/static/scripts/frontend_apps/components/test/JSTORPicker-test.js

```diff
@@ -1,26 +1,70 @@
 import { mount } from 'enzyme';
+import { useState } from 'preact/hooks';
 import { act } from 'preact/test-utils';
 
 import mockImportedComponents from '../../../test-util/mock-imported-components';
 import JSTORPicker, { $imports } from '../JSTORPicker';
 
 describe('JSTORPicker', () => {
-  let fakeToJSTORUrl;
+  let fakeArticleIdFromURL;
+
+  // Map of API path => fetch result state setter.
+  let setAPIFetchResult;
 
   function interact(wrapper, callback) {
     act(callback);
     wrapper.update();
   }
 
+  function simulateAPIFetch(wrapper, path, data, error = null) {
+    act(() => {
+      setAPIFetchResult[path]({
+        data,
+        error,
+        isLoading: false,
+      });
+    });
+    wrapper.update();
+  }
+
+  function simulateMetadataFetch(wrapper, title) {
+    simulateAPIFetch(wrapper, '/api/jstor/articles/1234', { title });
+  }
+
+  function simulateThumbnailFetch(wrapper) {
+    simulateAPIFetch(wrapper, '/api/jstor/articles/1234/thumbnail', {
+      image: 'data:thumbnail-image-data',
+    });
+  }
+
   const renderJSTORPicker = (props = {}) => mount(<JSTORPicker {...props} />);
 
   beforeEach(() => {
-    fakeToJSTORUrl = sinon.stub().returns('jstor://1234');
+    fakeArticleIdFromURL = sinon.stub().returns('1234');
+    setAPIFetchResult = {};
+
+    function useAPIFetchFake(path) {
```

Comment:
The mocking in this test module is wee bit complex...

Reply:
If we're going to mock stateful hooks, that is unavoidably a bit more fiddly than functions. We may be able to create or find a useful abstraction for this though.